### ISSUE TITLE
Typo in twisted.mail.mail.DomainWithDefaultDict

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1917]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#1917|
|Type|defect|
|Created|2006-07-12 14:13:26Z|

DomainWithDefaultDict is misspelled DomainWithDefaultsDict in the methods copy, str and repr.

Attachments:

* [mail_1917.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bc0/bc0d64ab9735781a3e0c72d96a7519674431c789/a952a4c55787c53725f8b970fe343f16e090c898.diff) (866 bytes) - added by therve on 2006-07-12 14:14:14Z - 
* [test_mail_1917.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/bc0/bc0d64ab9735781a3e0c72d96a7519674431c789/0ccd3f2b3e6020e623ac7f8bf7ca3577eabd5b2d.diff) (573 bytes) - added by therve on 2006-07-12 14:15:19Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1917 1917
type__defect defect
reporter__therve therve
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1152713606000000 1152713606000000
changetime__1186496471000000 1186496471000000
version__None None
owner__ 
cc__therve cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve commented|
|-|-|

Attached a path correcting the behaviour and one test for the copy method.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

I've been considering just deleting that class.  Do you know of any code which uses it directly?

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve commented|
|-|-|

No, I was just playing with MailService when I saw this problem. I guess there should be a simpler way to define the BounceDomain, so we can remove this.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

alright.  well, it's not even deprecated, so I can't delete it yet.  I applied your patch and made some further changes in domain-with-defaults-1917

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @exarkun|
|-|-|

 * _stringificationTest may benefit from a docstring
 * update copyright for mail.py

Then please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [20983]) Merge domain-with-defaults-1917

Author: exarkun, therve
Reviewer: therve
Fixes [#1917](https://github.com/twisted/twisted/issues/1917)

Fix several spelling errors in the domain support class in twisted.mail.mail
and add test coverage for the fixed methods.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
